File: 1051 Height Checker.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        heightfreq = [0] * 101;
        for height in heights:
            heightfreq[height] += 1;
        
        res = 0;
        currheight = 1;
        l = len(heights)
        
        for i in range(0, l):
            while heightfreq[currheight] == 0: currheight += 1;
            if currheight != heights[i]: res += 1;            
            heightfreq[currheight] -= 1;
        return res;

File: 1051 Height Checker.java
class Solution {
    public int heightChecker(int[] heights) {
        int[] heightfreq = new int[101];
        for (int height: heights) {
            heightfreq[height]++;
        }
        
        int res = 0;
        int currheight = 1;
        
        for (int i = 0; i < heights.length; ++i) {
            while (heightfreq[currheight] == 0) currheight++;
            if (currheight != heights[i]) res++;            
            heightfreq[currheight]--;
        }
        return res;
    }
}